src/core/pal/feature.cpp

```diff
@@ -642,117 +713,135 @@ std::size_t FeaturePart::createCandidatesAroundPoint( double x, double y, std::v
+    const double scaledRayAngleCost = RAY_ANGLE_COST_FACTOR * static_cast< double >( integerRayCost )
+                                      / static_cast< double >( rayCount - 1 );
+
+    for ( int j = 0; j < candidatesPerRay; ++j, rayDistance += rayStepDelta )
     {
-      deltaX = distanceToLabel;
-      double iota = ( angleToCandidate + gamma1 );
-      if ( iota > a360 - gamma1 )
-        iota -= a360;
+      LabelPosition::Quadrant quadrant = LabelPosition::QuadrantOver;
 
-      //ly += -yrm/2.0 + tan(alpha)*(distlabel + xrm/2);
-      deltaY = -labelHeight + labelHeight * iota / ( 2 * gamma1 );
+      if ( angleToCandidate < gamma1 || angleToCandidate > a360 - gamma1 )  // on the right
+      {
+        deltaX = rayDistance;
+        double iota = ( angleToCandidate + gamma1 );
+        if ( iota > a360 - gamma1 )
+          iota -= a360;
 
-      quadrant = LabelPosition::QuadrantRight;
-    }
-    else if ( angleToCandidate < a90 - gamma2 )  // top-right
-    {
-      deltaX = distanceToLabel * std::cos( angleToCandidate );
-      deltaY = distanceToLabel * std::sin( angleToCandidate );
-      quadrant = LabelPosition::QuadrantAboveRight;
-    }
-    else if ( angleToCandidate < a90 + gamma2 ) // top
-    {
-      //lx += -xrm/2.0 - tan(alpha+a90)*(distlabel + yrm/2);
-      deltaX = -labelWidth * ( angleToCandidate - a90 + gamma2 ) / ( 2 * gamma2 );
-      deltaY = distanceToLabel;
-      quadrant = LabelPosition::QuadrantAbove;
-    }
-    else if ( angleToCandidate < a180 - gamma1 )  // top left
-    {
-      deltaX = distanceToLabel * std::cos( angleToCandidate ) - labelWidth;
-      deltaY = distanceToLabel * std::sin( angleToCandidate );
-      quadrant = LabelPosition::QuadrantAboveLeft;
-    }
-    else if ( angleToCandidate < a180 + gamma1 ) // left
-    {
-      deltaX = -distanceToLabel - labelWidth;
-      //ly += -yrm/2.0 - tan(alpha)*(distlabel + xrm/2);
-      deltaY = - ( angleToCandidate - a180 + gamma1 ) * labelHeight / ( 2 * gamma1 );
-      quadrant = LabelPosition::QuadrantLeft;
-    }
-    else if ( angleToCandidate < a270 - gamma2 ) // down - left
-    {
-      deltaX = distanceToLabel * std::cos( angleToCandidate ) - labelWidth;
-      deltaY = distanceToLabel * std::sin( angleToCandidate ) - labelHeight;
-      quadrant = LabelPosition::QuadrantBelowLeft;
-    }
-    else if ( angleToCandidate < a270 + gamma2 ) // down
-    {
-      deltaY = -distanceToLabel - labelHeight;
-      //lx += -xrm/2.0 + tan(alpha+a90)*(distlabel + yrm/2);
-      deltaX = -labelWidth + ( angleToCandidate - a270 + gamma2 ) * labelWidth / ( 2 * gamma2 );
-      quadrant = LabelPosition::QuadrantBelow;
-    }
-    else if ( angleToCandidate < a360 ) // down - right
-    {
-      deltaX = distanceToLabel * std::cos( angleToCandidate );
-      deltaY = distanceToLabel * std::sin( angleToCandidate ) - labelHeight;
-      quadrant = LabelPosition::QuadrantBelowRight;
-    }
+        //ly += -yrm/2.0 + tan(alpha)*(distlabel + xrm/2);
```

Comment:
Are those serving a purpose in explaining the logic or should we get rid of them?

Reply:
I've just removed them. I don't think they add any value to anyone !